icepyx/core/granules.py

```diff
@@ -170,14 +175,19 @@ def __init__(
     # ----------------------------------------------------------------------
     # Methods
 
-    def get_avail(self, CMRparams, reqparams, cloud=False):
+    def get_avail(
+        self,
+        CMRparams: Optional[CMRParams],
+        reqparams: Optional[EGISpecificRequiredParams],
```

Comment:
None is not a valid input for these, so I don't think `Optional` is a valid type for them.

Reply:
I think we could actually limit `reqparams` further to `EGISpecificParamsSearch` too.

Reply:
Ooo, nice call, thank you!

Reply:
a8996d1